k2/csrc/fsa_algo.h

```diff
@@ -195,6 +195,41 @@ bool Intersect(FsaOrVec &a_fsas, FsaOrVec &b_fsas,
                bool treat_epsilons_specially, FsaVec *out,
                Array1<int32_t> *arc_map_a, Array1<int32_t> *arc_map_b);
 
+/*
+    Remove epsilons (symbol zero) in the input Fsas, it works for both
+    Fsa and FsaVec.
+    @param [in] src   Source Fsa or FsaVec. Must be top-sorted as we will
+                      compute forward and backward scores on it.
+    @param [out] dest Destination; at exit will be equivalent to `src` but will
+                      be epsilon-free, i.e. there's no arc in dest with
+                      arc.label==0, its Properties() will contain
+                      kFsaPropertiesEpsilonFree.
+    We don't output arc_map here as this version will not support auto grad; we
```

Comment:
autograd one word

Reply:
... also, equivalent in what semiring?  There were 2 versions of the host function.

Reply:
updated autogrid